### PR TITLE
CORDA-1641 - Add Java example to proxy serializer documentation

### PR DESCRIPTION
**Pull request by fenryka**
Another issue user interaction flagged is that the example code only had a Kotlin example... which they found initially offputting. Add a Java example to further clarify how things should be written.

### CODE REVIEW

File: docs/source/cordapp-custom-serializers.rst [hunk 1]
```diff
@@ -61,44 +67,120 @@ initialisation of all of its properties.
...
+             * These properties will be serialized into the byte stream, this is where we choose how to
+             * represent instances of the object we're proxying. In this example, which is somewhat
+             * contrived, this choice is obvious. In your own classes / 3rd party libraries, however, this
+             * may require more thought.
+             */
+            private Integer proxiedA;
+            private Integer proxiedB;
+
+            /**
+             * The proxu class itself must be serializable by the framework, it must thus have a constructor that
```
Comment (shamsasari) [inline docs/source/cordapp-custom-serializers.rst]: typo: proxy

File: docs/source/cordapp-custom-serializers.rst [hunk 2]
```diff
@@ -61,44 +67,120 @@ initialisation of all of its properties.
...
+         * with its properties and is thus not serializable by the CORDA serialization
+         * framework.
          */
-        data class Proxy(val a: Int, val b: Int)
-
+        static class Example {
+            private Integer a;
+            private Integer b;
+    
+            Integer getA() { return  a; }
```
Comment (shamsasari) [inline docs/source/cordapp-custom-serializers.rst]: public

File: docs/source/cordapp-custom-serializers.rst [hunk 3]
```diff
@@ -61,44 +67,120 @@ initialisation of all of its properties.
...
+            Integer getB() { return  b; }
```
Comment (shamsasari) [inline docs/source/cordapp-custom-serializers.rst]: ditto

File: docs/source/cordapp-custom-serializers.rst [hunk 4]
```diff
@@ -61,44 +67,120 @@ initialisation of all of its properties.
...
-         * This is the actual proxy class that is used as an intermediate representation
-         * of the Example class
+         * The class lacks a public constructor that takes parameters it can associate
+         * with its properties and is thus not serializable by the CORDA serialization
+         * framework.
          */
-        data class Proxy(val a: Int, val b: Int)
-
+        static class Example {
+            private Integer a;
```
Comment (shamsasari) [inline docs/source/cordapp-custom-serializers.rst]: Any particular reason to use `Integer`? `Integer` is rarely used in Java-land - more likely it's used where Objects are expected, e.g. list, sets, etc. More natural for this to be `int`s.

File: docs/source/cordapp-custom-serializers.rst [hunk 5]
```diff
@@ -61,44 +67,120 @@ initialisation of all of its properties.
...
-    }
 
-In the above ``ExampleSerializer`` is the actual serializer that will be loaded by the framework to
-serialize instances of the ``Example`` type.
+In the above examples
 
-``ExampleSerializer.Proxy`` is the intermediate representation used by the framework to represent
-instances of ``Example`` within the wire format.
+- ``ExampleSerializer`` is the actual serializer that will be loaded by the framework to serialize instances of the ``Example`` type.
+- ``ExampleSerializer.Proxy``, in the Kotlin example, and ``ExampleProxy`` in the Jaba example, is the intermediate representation used by the framework to represent instances of ``Example`` within the wire format.
```
Comment (shamsasari) [inline docs/source/cordapp-custom-serializers.rst]: Jaba-the-Hut!

File: docs/source/cordapp-custom-serializers.rst [hunk 6]
```diff
@@ -61,44 +67,120 @@ initialisation of all of its properties.
...
         /**
-         * This is the actual proxy class that is used as an intermediate representation
-         * of the Example class
+         * The class lacks a public constructor that takes parameters it can associate
+         * with its properties and is thus not serializable by the CORDA serialization
+         * framework.
          */
-        data class Proxy(val a: Int, val b: Int)
-
+        static class Example {
```
Comment (shamsasari) [inline docs/source/cordapp-custom-serializers.rst]: static classes in Java is required if doing nesting. But if `Example` is an example of a 3rd party lib class then it probably won't be an inner class.

File: docs/source/cordapp-custom-serializers.rst (hunk 1 above)
Comment (fenryka) [reply docs/source/cordapp-custom-serializers.rst]: done

File: docs/source/cordapp-custom-serializers.rst (hunk 4 above)
Comment (fenryka) [reply docs/source/cordapp-custom-serializers.rst]: not really, it just comes to mind when typing

File: docs/source/cordapp-custom-serializers.rst (hunk 5 above)
Comment (fenryka) [reply docs/source/cordapp-custom-serializers.rst]: done

**Pull request merged**